### PR TITLE
implement reveal in GameState and add various helper methods

### PR DESCRIPTION
**Pull request by michelleon**
mainly implements `reveal` but also adds helper methods to aid in supporting bureaucrat card

### CODE REVIEW

**Comment by michelleon** [comment]
ptal @jakemagner

File: core/events.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ class CardKnowledgeEventType(enum.Enum):
     'cards',
     'from_location',
     'from_position',
+    'number',
```
Comment (jakemagner) [inline core/events.py]: redundant info if cards are included

File: core/game_state.py [hunk 2]
```diff
@@ -251,14 +256,39 @@ def set_counter(self, counter_id, value):
...
+        cards = None
+        card_stack = self.get_location(location)
+        if number and location.type == LocationName.DRAW_PILE:
+            self.shuffle_discard_in_if_insufficient_cards(number)
+        if position and number:
+            cards = card_stack.peek_from_position(position, number)
+
+        self.logger.log(
+            CardKnowledgeEvent(
+                players,
```
Comment (jakemagner) [inline core/game_state.py]: Should we just include all the players? Reveal doesn't technically exclude the current player. Actually in some cases we must include the current player (like Rabble)

File: core/game_state.py [hunk 3]
```diff
@@ -251,14 +256,39 @@ def set_counter(self, counter_id, value):
...
+                cards or list(card_stack),
+                location,
+                position,
+                number or card_stack.size(),
```
Comment (jakemagner) [inline core/game_state.py]: number could be more than the number of cards revealed if there weren't enough cards

**Pull request merged**